webdriver/tests/bidi/storage/set_cookie/invalid.py

```diff
@@ -0,0 +1,152 @@
+import pytest
+from .. import create_cookie
+import webdriver.bidi.error as error
+from webdriver.bidi.modules.network import NetworkStringValue
+from webdriver.bidi.modules.storage import BrowsingContextPartitionDescriptor, StorageKeyPartitionDescriptor
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.mark.parametrize("domain", [None, False, 42, {}, []])
+async def test_cookie_domain_invalid_type(bidi_session, test_page, domain):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.storage.set_cookie(cookie=create_cookie(domain=domain))
+
+
+@pytest.mark.parametrize("expiry", [None, False, "SOME_STRING_VALUE", {}, []])
+async def test_cookie_expiry_invalid_type(bidi_session, test_page, domain_value, expiry):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.storage.set_cookie(cookie=create_cookie(domain=domain_value(), expiry=expiry))
+
+
+@pytest.mark.parametrize("http_only", [None, 42, "SOME_STRING_VALUE", {}, []])
+async def test_cookie_http_only_invalid_type(bidi_session, test_page, domain_value, http_only):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.storage.set_cookie(cookie=create_cookie(domain=domain_value(), http_only=http_only))
+
+
+@pytest.mark.parametrize(
+    "name",
+    [
+        " cookieName",
+        "cookie=name",
+        "cookie\tname",
+        "cookie\nname",
+        "cookie\x01name",
+        "cookie\x0Fname",
+        "cookie;name",
+    ])
+async def test_cookie_name_invalid_value(bidi_session, test_page, domain_value, name):
+    with pytest.raises(error.UnableToSetCookieException):
+        await bidi_session.storage.set_cookie(cookie=create_cookie(domain=domain_value(), name=name))
+
+
+@pytest.mark.parametrize("name", [None, False, 42, {}, []])
+async def test_cookie_name_invalid_type(bidi_session, test_page, domain_value, name):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.storage.set_cookie(cookie=create_cookie(domain=domain_value(), name=name))
+
+
+@pytest.mark.parametrize(
+    "path",
+    [
+        ""
+        "no_leading_forward_slash"
+    ]
+)
+async def test_cookie_path_invalid_value(bidi_session, test_page, domain_value, path):
+    with pytest.raises(error.UnableToSetCookieException):
+        await bidi_session.storage.set_cookie(
+            cookie=create_cookie(domain=domain_value(), path=path))
+
+
+@pytest.mark.parametrize("path", [None, False, 42, {}, []])
```

Comment:
I've just noticed that I think the introduction for cookies command usage of `Undefined` type breaks the existing pattern :( We usually for all the others command treat `undefined` and `null` as a not provided argument, so normally `None` here would a valid argument since it's an optional argument. We introduced `Undefined` type for `setViewport` command, where we have a special behavior for `null` value. So I think we have to revert it to not have special behavior just for these set of commands. Sorry for spotting it late :(

Reply:
In general, I find splitting `null` and `undefined` is a right thing to do, so I'd stick to this pattern in future tests. One obvious advantage - it allows us to test invalid `null` values. WDYT?

Reply:
But that's the thing, that would mean expecting a different behavior, at least for us, at the moment we treat `null` as `undefined` for optional arguments for all commands except `setViewport`.

Reply:
Reverted and file an issue to discuss: https://github.com/web-platform-tests/wpt/issues/44068

Reply:
UPD: https://github.com/w3c/webdriver-bidi/issues/642

Reply:
Thanks!